Sources/Spezi/Spezi/Spezi.swift

```diff
@@ -145,7 +146,8 @@ public final class Spezi: Sendable { // swiftlint:disable:this type_body_length
         }
     }
     
-    @MainActor var modules: [any Module] {
+    @_spi(APISupport)
```

Comment:
Have we ever documented the difference between APISupport and the Spezi SPI targets? I am wondering if some of them are now obsolete with the fact that we can have `package` visibility?

We should at least document this somewhere within the package for future usage and differentiating between the two.

Reply:
no but the package visibility thing is a good idea; i'll look into that. 

my understanding was that "Spezi" is used for APIs that need to be exposed to other parts of the core Spezi infrastructure (ie, what we might be able to replace with package visibility), and "APISupport" is used in cases where we need to expose implementation details to some other (non-core) package (to support the implement of that package's API), but don't want to make it part of Spezi's public API

Reply:
had a look; we only import `@_spi(Spezi)` definitions in the Spezi repo itself (apart from one file in SpeziBluetooth, where it actually can be removed bc it isn't actually needed). since all of these imports are in the testing-related targets, we can probably replace them with `package` visibility; i'll have a look.

sadly, this approach will only work for definitions that are used directly within the package, i.e. it will probably not work for stuff the UITests need to access.

Reply:
@PSchmiedmayer was able to remove the "Spezi" SPI entirely (and switch everything over to `package` visibility). we now only have `@_spi(APISupport)`, which is used in a couple of places.

Reply:
Sounds great; nice improvement!

I think that's fine, we could add an API surface within this package as a separate target or as part of SpeziTesting that could expose anything targeted in the future if needed (I don't see a specific use case now).